src/map/pc.cpp

```diff
@@ -1972,6 +1972,47 @@ bool pc_set_hate_mob(struct map_session_data *sd, int pos, struct block_list *bl
 	return true;
 }
 
+TIMER_FUNC(pc_goldpc_update){
+	struct map_session_data* sd = map_id2sd( id );
+
+	if( sd == nullptr ){
+		return 0;
+	}
+
+	sd->goldpc_tid = INVALID_TIMER;
+
+	// Check if feature is still active
+	if( !battle_config.feature_goldpc_active ){
+		return 0;
+	}
+
+	// TODO: add mapflag to disable?
+
+	int32 points = (int32)pc_readreg2( sd, GOLDPC_POINT_VAR );
```

Comment:
You must change pc_readreg2 to a variable similar to Cash Point.

	int32 points = sd->goldpcPoints;

	if( points < battle_config.feature_goldpc_max_points ){
		if( battle_config.feature_goldpc_vip && pc_isvip( sd ) ){
			points += 2;
		}else{
			points += 1;
		}

		points = std::min( points, battle_config.feature_goldpc_max_points );

		sd->goldpcPoints = points;
		pc_setreg2( sd, GOLDPC_POINT_VAR, sd->goldpcPoints );
		pc_setreg2( sd, GOLDPC_SECONDS_VAR, 0 );

		if( points < battle_config.feature_goldpc_max_points ){
			sd->goldpc_tid = add_timer( gettick() + battle_config.feature_goldpc_time * 1000, pc_goldpc_update, sd->bl.id, (intptr_t)nullptr );
		}

		// Update the client
		clif_goldpc_info( *sd );
	}